### ISSUE TITLE
Plugin interaction and SkyWalking native log format with trace context

### ISSUE THREAD

**Issue by wu-sheng** [apache/apisix#4232]
@dmsolr @moonming I want to discuss an enhancement feature for APISIX ecosystem about observability.

SkyWalking built its native tracer implementation for Nginx and APISIX, so we have this, http://apisix.apache.org/docs/apisix/plugins/skywalking/. Which supports tracing and backed by OAP's analysis capability, we have topology with metrics about traffic.

Now, with the latest [SkyWalking's log analyzer](https://skywalking.apache.org/docs/main/v8.5.0/en/setup/backend/log-analyzer/), we should consider looping logs in. So 2 questions should be answered

1. Could 2 plugins interact with each other? If so, how.
2. Does APISIX want to adopt SkyWalking native log format? https://github.com/apache/skywalking-data-collect-protocol/blob/master/logging/Logging.proto Although APISIX is only good at HTTP+JSON, so we could leverage satellite to forward, but it is better to add a native gRPC reporter into satellite

If you feel interests, let's work on this together.

**Comment by wu-sheng** [comment]
I know we talked about ALS, which could be another separate feature. As general logs are more flexible, ALS is focusing on topology and traffic metrics analysising.

**Comment by dmsolr** [comment]
1. `plugins interact` means passing trace context in the scope of a request. We can use `ngx.ctx` to propagate it.
2. I think it is good to adopt SkyWalking native log format when `skywalking tracer plugin` was enabled.

**Comment by wu-sheng** [comment]
> plugins interact means passing trace context in the scope of a request. We can use ngx.ctx to propagate it.

Yes, it could be, but we should take care of the priority and execution sequence.

**Comment by wu-sheng** [comment]
> I think it is good to adopt SkyWalking native log format when skywalking tracer plugin was enabled.

I think we could include it in skywalking-nginx-lua repo if that makes senses to APISIX community.

**Comment by dmsolr** [comment]
APISIX has supported reporting access-log with trace context and error-log to SkyWalking by #5478 and #4633.
So, I think this issue could be closed.

**Comment by wu-sheng** [comment]
I think (1) is not supported? Do we skip it?

**Comment by dmsolr** [comment]
We can get `tracing_context` in other plugins by `ngx.ctx.tracing_context`. So I think (1) is supported as well.
I will submit a PR to add some explanations to the related document.

**Comment by wu-sheng** [comment]
Thanks, look forward to that.

**Comment by Baoyuantop** [comment]
The feature mentioned in this issue has been implemented and I believe it can be disabled. Please feel free to reopen it if you have any questions.
